uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js

```diff
@@ -0,0 +1,419 @@
+  'tasks-assignments/services/dialog',
+  'tasks-assignments/services/document',
+  'common/services/notification'
+], function (angular, moment, _, controllers) {
+  'use strict';
+
+  controllers.controller('ModalDocumentCtrl', ModalDocumentCtrl);
+
+  ModalDocumentCtrl.$inject = [
+    '$scope', '$uibModalInstance', '$rootScope', '$rootElement', '$q', '$log', 'role',
+    '$filter', '$uibModal', '$dialog', '$timeout', 'AssignmentService', 'DocumentService',
+    'ContactService', 'FileService', 'data', 'files', 'config', 'HR_settings', 'modalMode',
+    'notification'
+  ];
+
+  function ModalDocumentCtrl ($scope, $modalInstance, $rootScope, $rootElement, $q, $log, role,
+    $filter, $modal, $dialog, $timeout, AssignmentService, DocumentService, ContactService,
+    FileService, data, files, config, HRSettings, modalMode, notification) {
+    $log.debug('Controller: ModalDocumentCtrl');
+
+      // Init call
+    (function init () {
+      $scope.files = [];
+      $scope.document = {};
+
+      angular.copy(data, $scope.document);
+      angular.copy(files, $scope.files);
+
+      $scope.data = data;
+      $scope.role = role || 'admin';
+      $scope.modalTitle = modalMode === 'edit' ? 'Edit Document' : 'New Document';
+
+      $scope.document.activity_date_time = $scope.document.activity_date_time ? moment($scope.document.activity_date_time).toDate() : null;
+      $scope.document.expire_date = $scope.document.expire_date ? moment($scope.document.expire_date).toDate() : null;
+      $scope.document.valid_from = $scope.document.valid_from ? moment($scope.document.valid_from).toDate() : null;
+      $scope.document.assignee_contact_id = $scope.document.assignee_contact_id || [];
+      $scope.document.source_contact_id = $scope.document.source_contact_id || config.LOGGED_IN_CONTACT_ID;
+      $scope.document.target_contact_id = $scope.document.target_contact_id || [config.CONTACT_ID];
+      $scope.document.status_id = $scope.document.status_id || '1';
+
+      $scope.filesTrash = [];
+      $scope.uploader = FileService.uploader('civicrm_activity');
+      $scope.showCId = !config.CONTACT_ID;
+
+      $scope.assignments = $filter('filter')($rootScope.cache.assignment.arrSearch, function (val) {
+        return +val.extra.contact_id === +$scope.document.target_contact_id;
+      });
+
+      $scope.contacts = {
+        target: initialContacts('target'),
+        assignee: initialContacts('assignee')
+      };
+
+      $scope.dpOpened = {
+        due: false,
+        exp: false,
+        form: false
+      };
+
+      $scope.remindMeMessage = 'Checking this box sets a reminder that this document needs to be renewed a set number of days before the Expiry Date. You can set this by going <a target="_blank" href="/civicrm/tasksassignments/settings">here</a> CiviHR will do this by creating a copy of this document with the status ‘awaiting upload’, which you will be able to see in your Documents list.';
+    })();
+
+      /**
```

Comment:
Something wrong with comment indentation across the whole file. Please fix if not yet fixed in another PR.